backend/store/deploy.go

```diff
@@ -242,3 +242,155 @@ func (*Store) upsertDeploymentConfigImpl(ctx context.Context, tx *Tx, upsert *ap
 	}
 	return &cfg, nil
 }
+
+// UpsertDeploymentConfigMessage is the message to upsert a deployment config.
+type UpsertDeploymentConfigMessage struct {
+	ProjectUID       int
+	PrincipalUID     int
+	DeploymentConfig *DeploymentConfigMessage
+}
+
+// DeploymentConfigMessage is the message for deployment config.
+type DeploymentConfigMessage struct {
+	Name     string
+	Schedule *Schedule
+}
+
+// Schedule is the message for deployment schedule.
+type Schedule struct {
+	Deployments []*Deployment `json:"deployments"`
+}
+
+// Deployment is the message for deployment.
+type Deployment struct {
+	Name string          `json:"name"`
+	Spec *DeploymentSpec `json:"spec"`
+}
+
+// DeploymentSpec is the message for deployment specification.
+type DeploymentSpec struct {
+	Selector *LabelSelector `json:"selector"`
+}
+
+// LabelSelector is the message for label selector.
+type LabelSelector struct {
+	// MatchExpressions is a list of label selector requirements. The requirements are ANDed.
+	MatchExpressions []*LabelSelectorRequirement `json:"matchExpressions"`
+}
+
+// OperatorType is the type of label selector requirement operator.
+// Valid operators are In, Exists.
+// Note: NotIn and DoesNotExist are not supported initially.
+type OperatorType string
+
+const (
+	// InOperatorType is the operator type for In.
+	InOperatorType OperatorType = "In"
+	// ExistsOperatorType is the operator type for Exists.
+	ExistsOperatorType OperatorType = "Exists"
+)
+
+// LabelSelectorRequirement is the message for label selector.
+type LabelSelectorRequirement struct {
+	// Key is the label key that the selector applies to.
+	Key string `json:"key"`
+
+	// Operator represents a key's relationship to a set of values.
+	Operator OperatorType `json:"operator"`
+
+	// Values is an array of string values. If the operator is In or NotIn, the values array must be non-empty. If the operator is Exists or DoesNotExist, the values array must be empty. This array is replaced during a strategic merge patch.
+	Values []string `json:"values"`
+}
+
+// GetDeploymentConfigV2 returns the deployment config.
```

Comment:
Not necessary in this PR but we need to return the default deployment config and cache things.